starfish/core/types/_spot_finding_resutls.py

```diff
@@ -12,7 +15,7 @@ class SpotFindingResults:
     SpotAttributes.
     """
 
-    def __init__(self, spot_attributes_list: Optional[List[Tuple]] = None):
+    def __init__(self, imagestack, spot_attributes_list: Optional[List[Tuple]] = None):
```

Comment:
is it plausible that datasets get so big that we need to spread spot finding across multiple machines?  If so, having the imagestack as part of spotfinding's results may make it challenging to persist.

It _might_ make sense to store the coordinates from the imagestack, but not necessarily the actual imagestack (as that's really what you need anyway).

@ambrosejcarr, @dganguli